protocols/identify/src/behaviour.rs

```diff
@@ -471,6 +471,7 @@ impl PeerCache {
             Some(cache) => cache,
         };
 
+        let addresses = addresses.filter_map(|a| a.with_p2p(peer).ok());
```

Comment:
I find the new `with_p2p` very ergonomic.

Reply:
Yeah I agree. Our brainstorming with Thomas was definitely worth it :blush: